sdk/identity/azure-identity/azure/identity/aio/_credentials/base.py

```diff
@@ -0,0 +1,28 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import abc
+
+
+class AsyncCredentialBase(abc.ABC):
+    @abc.abstractmethod
+    async def close(self):
+        pass
+
+    @abc.abstractmethod
+    async def __aenter__(self):
+        pass
+
+    async def __aexit__(self, *args):
+        await self.close()
+
+    def __enter__(self):
+        raise TypeError("Use async with instead")
+
+    def __exit__(self, *args):
```

Comment:
Why implement `__exit__` if `__enter__` always throws?

Reply:
I anticipated a linting error for `__enter__` without `__exit__`. I don't get one though, so I'll remove `__exit__`.